contracts/test/integration/IntegrationTestMakerRouter.t.sol

```diff
@@ -0,0 +1,118 @@
+    IDSSPSM private daiPSM = IDSSPSM(MainnetAddresses.MAKER_DAI_USDC_PSM);
+    PegStabilityModule private feiPSM =
+        PegStabilityModule(MainnetAddresses.FEI_DAI_PSM);
+
+    IERC20 private dai = IERC20(MainnetAddresses.DAI);
+    IVolt private fei = IVolt(MainnetAddresses.FEI);
+    IERC20 private usdc = IERC20(MainnetAddresses.USDC);
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    uint256 public constant mintAmount = 200_000_000e18;
+
+    function setUp() public {
+        makerRouter = new MakerRouter(
+            MainnetAddresses.CORE,
+            daiPSM,
+            feiPSM,
+            dai,
+            fei
+        );
+
+        fei.approve(address(makerRouter), type(uint256).max);
+        dai.approve(address(makerRouter), type(uint256).max);
+
+        vm.prank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(feiPSM), mintAmount + 100_000_000e18);
+
+        vm.prank(MainnetAddresses.FEI_GOVERNOR);
+        fei.mint(address(this), mintAmount);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantGovernor(address(this));
+    }
+
+    function testSwapFeiForDai(uint64 amountFeiIn) public {
+        vm.assume(amountFeiIn > 1e18);
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(amountFeiIn);
+        makerRouter.swapFeiForDai(amountFeiIn, minDaiAmountOut, address(this));
+
+        assertEq(minDaiAmountOut, dai.balanceOf(address(this)));
+    }
+
+    function testSwapFeiForUsdc(uint64 amountFeiIn) public {
+        vm.assume(amountFeiIn > 1e18);
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(amountFeiIn);
+        makerRouter.swapFeiForUsdc(amountFeiIn, minDaiAmountOut, address(this));
+
+        assertEq((minDaiAmountOut / 1e12), usdc.balanceOf(address(this)));
+    }
+
+    function testSwapFeiForUsdcAndDai(uint64 amountFeiIn, uint8 ratioUSDC)
+        public
+    {
+        vm.assume(amountFeiIn > 1e18);
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(amountFeiIn);
+        makerRouter.swapFeiForUsdcAndDai(
+            amountFeiIn,
+            minDaiAmountOut,
+            ratioUSDC,
+            address(this)
+        );
+
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) / 10000;
+
+        assertEq((minDaiAmountOut - usdcAmount), dai.balanceOf(address(this)));
+        assertEq((usdcAmount / 1e12), usdc.balanceOf(address(this)));
+    }
+
+    function testSwapAllFeiForDai() public {
+        makerRouter.swapAllFeiForDai(address(this));
+
+        assertEq(
+            dai.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount)
+        );
+    }
+
+    function testSwapAllFeiForUsdc() public {
+        makerRouter.swapAllFeiForUsdc(address(this));
+
+        assertEq(
+            usdc.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount) / 1e12
+        );
+    }
+
+    function testSwapAllFeiForUsdcAndDai(uint8 ratioUSDC) public {
```

Comment:
this test will only test with a max ratioUSDC of 255, we could consider using uint16, and assuming the input is less than 10k, or add a separate test that just tests a value above 255 by hand.